providers/termii/src/lib/termii.provider.ts

```diff
@@ -58,4 +60,60 @@ export class TermiiSmsProvider implements ISmsProvider {
       date: new Date().toISOString(),
     };
   }
+  getMessageId(body: any | any[]): string[] {
+    if (Array.isArray(body)) {
+      return body.map((item) => item.message_id);
+    }
+
+    return [body.message_id];
+  }
+
+  parseEventBody(
+    body: any | any[],
+    identifier: string
+  ): ISMSEventBody | undefined {
+    if (Array.isArray(body)) {
+      body = body.find((item) => item.message_id === identifier);
+    }
+
+    if (!body) {
+      return undefined;
+    }
+
+    const status = this.getStatus(body.status);
+
+    if (status === undefined) {
+      return undefined;
+    }
+
+    return {
+      status: status,
+      date: new Date().toISOString(),
+      externalId: body.message_id,
+      attempts: body.attempt ? parseInt(body.attempt, 10) : 1,
+      response: body.response ? body.response : '',
+      row: body,
+    };
+  }
+
+  private getStatus(event: string): SmsEventStatusEnum | undefined {
+    switch (event) {
+      case 'DND Active on Phone Number':
+        return SmsEventStatusEnum.DND_ACTIVE;
+      case 'Expired':
+        return SmsEventStatusEnum.EXPIRED;
+      case 'Message sent':
+        return SmsEventStatusEnum.SENT;
+      case 'Message failed':
+        return SmsEventStatusEnum.FAILED;
+      case 'Delivered':
+        return SmsEventStatusEnum.DELIVERED;
+      case 'Disconnected':
+        return SmsEventStatusEnum.DISCONNECTED;
+      case 'Rejected':
+        return SmsEventStatusEnum.REJECTED;
+      case 'Received':
+        return SmsEventStatusEnum.RECEIVED;
+    }
```

Comment:
@djabarovgeorge after what we discussed the other day which approach should we recommend to the contributor? I still find having hardcoded cases to be very brittle but that would mean this provider package wouldn't be agnostic to Novu.
@scopsy @davidsoderberg @ainouzgali this is a topic worth to be discussed inside the team.

Reply:
I would like to hear others opinion on the matter.